### ISSUE TITLE
RN Windows Support?

### ISSUE THREAD

**Issue by gmacmaster** [mrousavy/nitro#168]
I see mention on iOS and Android in the doc but no mention of [RNW](https://microsoft.github.io/react-native-windows/). 

Is there a way to get this working on windows?

**Comment by mrousavy** [comment]
To add support to another platform, you just need to call `margelo::nitro::install(...)` in a Runtime. This will install `global.NitroModulesProxy` into the Runtime's global.

https://github.com/mrousavy/nitro/blob/c0ae25bef6710894c1314f714638b38ca650c66b/packages/react-native-nitro-modules/cpp/entrypoint/InstallNitro.hpp#L16-L29

cc @shirakaba

**Comment by shirakaba** [comment]
[Here](https://github.com/microsoft/react-native-windows/discussions/13886#discussioncomment-10723912) is my writeup of how to write a `react-native-windows` attributed module (which can be installed as either a NativeModule or TurboModule).

I'd help out, but I don't have even a spare half-hour from now until mid-November 🫠

**Comment by kziemski** [comment]
this would be useful to look at again with nitro adoption in a bunch of places. 
will take a look at building a pr based on on the above. 
bookmarking this. love vision btw ty for the hard work.

**Comment by juhasuni** [comment]
@kziemski , did you have a chance to look into this yet? We're also interested in this development.

**Comment by tux2nicolae** [comment]
Any update on this? I see https://github.com/StasDoskalenko/NitromelonDB/pull/65 managed to workaround this, but it would be nice to have a native support for react-native-windows here.

**Comment by StasDoskalenko** [comment]
Hey @tux2nicolae thanks for mentioning my repo 🤝 

I worked around this in NitromelonDB ([#65](https://github.com/StasDoskalenko/NitromelonDB/issues/65)). The `react-native-nitro-modules` JS still calls `TurboModuleRegistry.getEnforcing('NitroModules').install()` like on iOS/Android.

On Windows I implement that native module in WinAppSDK DLL:
1. Skip Nitro’s missing Windows project in the app `react-native.config.js` (`platforms.windows: null`) so RNW does not look for a vcxproj this repo does not ship. Helper: [`windows-autolink.js`](https://github.com/StasDoskalenko/NitromelonDB/blob/master/windows-autolink.js).
2. Compile Nitro C++ from `node_modules/react-native-nitro-modules/cpp` into my DLL. Source list is in [`NitromelonDB.vcxproj`](https://github.com/StasDoskalenko/NitromelonDB/blob/master/native/windows/NitromelonDB/NitromelonDB.vcxproj).
3. Implement the `NitroModules` TurboModule myself (`REACT_MODULE(NitroModules)` → `margelo::nitro::install()` + register `HybridNitromelon`). [`NitromelonDB.cpp`](https://github.com/StasDoskalenko/NitromelonDB/blob/master/native/windows/NitromelonDB/NitromelonDB.cpp).
4. Generate `<NitroModules/Foo.hpp>` shims for MSVC (iOS/Android get that prefix from a header map). [`scripts/windows-nitro-shims.mjs`](https://github.com/StasDoskalenko/NitromelonDB/blob/master/scripts/windows-nitro-shims.mjs) (yarn postinstall and an MSBuild target).


Writeup: [`native/windows/README.md`](https://github.com/StasDoskalenko/NitromelonDB/blob/master/native/windows/README.md).

Happy to help land this in nitro itself. What should live here: a Windows vcxproj (or CMake) that compiles `cpp/`, a `REACT_MODULE(NitroModules)` install entry, autolink in `react-native.config.js`, and an MSVC header map (or generated shims). 

Once that exists it would be so much easier to use 👍

**Comment by mrousavy** [comment]
That's great stuff! Can we test react-native-windows with Harness as well? I don't have a Windows dev machine so maintenance has to be light and run entirely in CI

**Comment by StasDoskalenko** [comment]
@mrousavy I started to work on this [in my fork](https://github.com/StasDoskalenko/nitro). When I have something testable, I'll start opening PRs :) 
A few things to note:

1. react-native-windows@^0.84 is at react-native 0.84. I'm thinking about an example app. What if I create an example_windows folder so there are no RN version collisions?
2. Harness doesn't support windows :( However, I think we can fix that.

**Comment by StasDoskalenko** [comment]
Quick update, I think I managed to expand Harness support to Windows...
https://github.com/StasDoskalenko/nitro/pull/8
https://github.com/StasDoskalenko/nitro/pull/11 

I'll open a PR against harness to avoid patching it from windows PR in this repo :D }


upd: https://github.com/callstackincubator/react-native-harness/pull/187

**Comment by tux2nicolae** [comment]
@StasDoskalenko The https://github.com/callstackincubator/react-native-harness/releases was just released yesterday, can you help please in unblocking this PR?

**Comment by StasDoskalenko** [comment]
Hey @tux2nicolae thanks for tagging me!
I already see a PR here https://github.com/margelo/nitro/pull/1649
As soon as it's merged, I'll rebase my fork and open a PR with windows support 👍
